great_expectations/data_context/store/validation_config_store.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from great_expectations.compatibility.typing_extensions import override
+from great_expectations.core.data_context_key import StringKey
+from great_expectations.core.validation_config import ValidationConfig
+from great_expectations.data_context.cloud_constants import GXCloudRESTResource
+from great_expectations.data_context.store.store import Store
+from great_expectations.data_context.types.resource_identifiers import (
+    GXCloudIdentifier,
+)
+
+
+class ValidationConfigStore(Store):
+    _key_class = StringKey
+
+    def get_key(
+        self, name: str, id: str | None = None
+    ) -> GXCloudIdentifier | StringKey:
+        """Given a name and optional ID, build the correct key for use in the ValidationConfigStore."""
+        if self.cloud_mode:
```

Comment:
Not for this PR but we need to get away from these cloud mode checks.

Reply:
I agree. It's poorly encapsulated but I couldn't think of a better way to get both file and cloud working